typeclasses/chargen.py

```diff
@@ -0,0 +1,587 @@
+        output = "Final Skills:\n"
+        output += "{skills}\n"
+        output += "You get |w{coins}|n SC (Silver Coins) to start out.\n"
+
+        char.db.wallet['SC'] = d_roll('2d6+3')
+
+        return menunode_equipment_cats(
+            caller,
+            output.format(skills=table,
+                          coins=char.db.wallet['SC'])
+        )
+
+
+def menunode_equipment_cats(caller, raw_input):
+    """Initial equipment "shopping" - choose a category"""
+    text = raw_input if raw_input and raw_input[0] == 'F' else ""
+    text += "Select a category of equipment to view."
+
+    def show_inventory(s):
+        """display the character's inventory"""
+        s.msg('\n')
+        old_msg = s.new_char.msg
+        s.new_char.msg = s.msg
+        s.new_char.execute_cmd('inventory')
+        s.new_char.msg = old_msg
+
+    options = [{"desc": cat, "goto": "menunode_equipment_list"}
+               for cat in _CATEGORY_LIST]
+
+    options.append({"key": ("Inventory", "inv", "i"),
+                    "desc": "Show your current inventory",
+                    "exec": show_inventory,
+                    "goto": "menunode_equipment_cats"})
+
+    options.append({"key": "Done",
+                    "desc": "Continue to character description",
+                    "goto": "menunode_character_desc"})
+
+    return (text, help), options
+
+
+def menunode_equipment_list(caller, raw_input):
+    """Initial equipment "shopping" - list items in a category"""
+    text = "Select an item to view details and buy."
+
+    if raw_input.isdigit() and int(raw_input) <= len(_CATEGORY_LIST):
+        caller.ndb._menutree.item_category = _CATEGORY_LIST[int(raw_input) - 1]
+
+    category = (caller.ndb._menutree.item_category
+                if hasattr(caller.ndb._menutree, 'item_category')
+                else '')
+
+    prototypes = spawn(return_prototypes=True)
+    options = []
+    for proto in _EQUIPMENT_CATEGORIES[category][1:]:
+        options.append({
+            "desc": _format_menuitem_desc(prototypes[proto]),
+            "goto": "menunode_examine_and_buy"
+        })
+    options.append({
+        "key": ("Back", "_default"),
+        "desc": "to category list",
+        "goto": "menunode_equipment_cats",
+    })
+    return text, options
+
+
+def menunode_examine_and_buy(caller, raw_input):
+    """Examine and buy an item."""
+    char = caller.new_char
+    prototypes = spawn(return_prototypes=True)
+    items, item = _EQUIPMENT_CATEGORIES[caller.ndb._menutree.item_category][1:], None
+    if raw_input.isdigit() and int(raw_input) <= len(items):
+        item = prototypes[items[int(raw_input)-1]]
+    if item:
+        text = _format_item_details(item)
+        text += "You currently have {}. Purchase |w{}|n?".format(
+                    as_price(char.db.wallet),
+                    item['key']
+                )
+        help = "Choose carefully. Purchases are final."
+
+        def purchase_item(caller):
```

Comment:
perhaps` purchase_item` could have a `storeroom` boolean that specified the behavior? The potential repeated code doesn't feel right.